src/scripts/extensions/safari/safariWorker.ts

```diff
@@ -129,34 +129,37 @@ export class SafariWorker extends ExtensionWorkerBase<SafariBrowserTab, SafariBr
 		});
 	}
 
-	private launchSafariPopupAndWaitForClose(url: string, autoCloseDestinationUrl: string): Promise<boolean> {
+	private launchSafariPopup(url: string, autoCloseDestinationUrl: string, waitForClose = false): Promise<boolean> {
 		return new Promise<boolean>((resolve, reject) => {
-			let redirectOccurred = false;
 			let newWindow = safari.application.openBrowserWindow();
 			newWindow.activeTab.url = url;
 
-			let errorObject;
-			let redirectListener = (event) => {
-				if (event && event.target && event.target.url && !event.target.url.toLowerCase().indexOf(autoCloseDestinationUrl)) {
-					redirectOccurred = true;
+			if (waitForClose) {
+				let redirectOccurred = false;
 
-					let error = Utils.getQueryValue(event.target.url, Constants.Urls.QueryParams.error);
-					let errorDescription = Utils.getQueryValue(event.target.url, Constants.Urls.QueryParams.errorDescription);
-					if (error || errorDescription) {
-						errorObject = { error: error, errorDescription: errorDescription };
-					}
+				let errorObject;
+				let redirectListener = (event) => {
+					if (event && event.target && event.target.url && !event.target.url.toLowerCase().indexOf(autoCloseDestinationUrl)) {
+						redirectOccurred = true;
+
+						let error = Utils.getQueryValue(event.target.url, Constants.Urls.QueryParams.error);
+						let errorDescription = Utils.getQueryValue(event.target.url, Constants.Urls.QueryParams.errorDescription);
+						if (error || errorDescription) {
+							errorObject = { error: error, errorDescription: errorDescription };
+						}
 
-					newWindow.removeEventListener(redirectListener);
-					newWindow.close();
-				}
-			};
-			newWindow.addEventListener("navigate", redirectListener);
-
-			let closeListener = (event) => {
-				errorObject ? reject(errorObject) : resolve(redirectOccurred);
-				newWindow.removeEventListener(closeListener);
-			};
-			newWindow.addEventListener("close", closeListener);
+						newWindow.removeEventListener(redirectListener);
+						newWindow.close();
```

Comment:
Shouldn't we wrap the new if statement around the .close() call? That way we still get our error reporting.